### DISCUSSION TITLE
Default Outbound Proxy Connect Timeout

### DISCUSSION THREAD

**Discussion by ngc4579** [linkerd/linkerd2#10952] [Q&A] [answered]
What is the default outbound proxy connect timeout (configurable as `config.linkerd.io/proxy-outbound-connect-timeout`? I couldn't find documentation on that...

**Comment by mateiidavid** [comment, marked as answer]
Hi @ngc4579, default outbound connect timeout is 1s. An easy way to verify this is to look at the pod's environment:

```sh
$ kubectl get po -n linkerd linkerd-destination-596d54dfb4-jxq6v -o yaml \
| rg 'LINKERD2_PROXY_OUTBOUND_CONNECT_TIMEOUT' -A 1
    - name: LINKERD2_PROXY_OUTBOUND_CONNECT_TIMEOUT
      value: 1000ms
```


(also, see [code ref](https://github.com/linkerd/linkerd2-proxy/blob/704ef31a286280187c1f4679170e897243f99900/linkerd/app/src/env.rs#L252))

**Reply by ngc4579** [reply]
Thank you so much for the explanation @mateiidavid. I've been struggling with timeouts on outgoing connections in some pods and was looking for potential configuration on Linkerd's side that might possibly be causing those.

In fact I have some Linkerd meshed pods sending traffic to an external service, and all of those requests are capped after 30s, i.e. run into a timeout. Currently I've no idea what is causing this. I keep getting a "timeout" error message inside my application, yet I suspect the response would be some kind of "connection reset by peer" in case the timeout was caused by the target service, so at the moment my guess is something imposes a max connection read / write timeout of 30s inside my application.

Is there any such configuration for read or write timeouts inside Linkerd?
